### PR TITLE
alliance-36 Fixes for exceptions when attempting to retrieve results.

### PR DESCRIPTION
**Pull request by troymohl**
#### What does this PR do?
Fixes exceptions that occur when attempting to retrieve hit counts and results against a STANAG-4559 server.
#### Who is reviewing it?
@jaymcnallie @kcwire @bdeining @jlcsmith 
#### How should this be tested?
Build and run unit tests
#### What are the relevant tickets?
https://github.com/connexta/alliance/issues/36
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/nsili/catalog-nsili-transformer/src/main/java/com/connexta/alliance/nsili/transformer/DAGConverter.java [hunk 1]
```diff
@@ -1143,29 +1143,23 @@ private static String collectionToString(Collection<String> collection) {
...
-                metacardType.getClass()
-                        .getCanonicalName());
-        LOGGER.trace("{} :  Title : {}", id, metacard.getTitle());
+        LOGGER.trace("{} : ID : " + metacard.getId(), id);
+        LOGGER.trace("{} :  Metacard Type : " + metacardType.getClass()
+                .getCanonicalName(), id);
+        LOGGER.trace("{} :  Title : " + metacard.getTitle(), id);
         if (metacard instanceof MetacardImpl) {
-            LOGGER.trace("{} :  Description : {}", id, ((MetacardImpl) metacard).getDescription());
+            LOGGER.trace("{} :  Description : " + ((MetacardImpl) metacard).getDescription(), id);
```
Comment (bdeining) [inline catalog/nsili/catalog-nsili-transformer/src/main/java/com/connexta/alliance/nsili/transformer/DAGConverter.java]: Keep this parameterized

Comment (troymohl) [inline catalog/nsili/catalog-nsili-transformer/src/main/java/com/connexta/alliance/nsili/transformer/DAGConverter.java]: That change is part of the CAL-14 / CAL-17 PR. One of them will need to be rebased depending on what gets merged first.

**Comment by bdeining** [comment]
👍

File: catalog/nsili/catalog-nsili-transformer/src/main/java/com/connexta/alliance/nsili/transformer/DAGConverter.java [hunk 2]
```diff
@@ -1171,9 +1171,11 @@ public static void logMetacard(Metacard metacard, String id) {
             if (attribute != null) {
                 if (attribute.getValues() != null) {
                     String valueStr = getValueString(attribute.getValues());
-                    LOGGER.trace("{} :  {} : {}", id, descriptor.getName(), valueStr);
+                    LOGGER.trace("{} :  {}", id, descriptor.getName() + " : " +
```
Comment (bdeining) [inline catalog/nsili/catalog-nsili-transformer/src/main/java/com/connexta/alliance/nsili/transformer/DAGConverter.java]: Don't have to concatenate  :  `LOGGER.trace("{} :  {} : {}", id, descriptor.getName(), valueStr);

File: catalog/nsili/catalog-nsili-transformer/src/main/java/com/connexta/alliance/nsili/transformer/DAGConverter.java [hunk 3]
```diff
@@ -1171,9 +1171,11 @@ public static void logMetacard(Metacard metacard, String id) {
...
+                            valueStr);
                 } else {
-                    LOGGER.trace("{} : {} : {}", id, descriptor.getName(), attribute.getValue());
+                    LOGGER.trace("{} :  {}", id, descriptor.getName() + " : " +
```
Comment (bdeining) [inline catalog/nsili/catalog-nsili-transformer/src/main/java/com/connexta/alliance/nsili/transformer/DAGConverter.java]: Same comment

Comment (troymohl) [inline catalog/nsili/catalog-nsili-transformer/src/main/java/com/connexta/alliance/nsili/transformer/DAGConverter.java]: Fixed both places.

File: catalog/nsili/catalog-nsili-source/src/main/java/com/connexta/alliance/nsili/source/NsiliSource.java [hunk 4]
```diff
@@ -471,6 +472,7 @@ private void initSortableAndResultAttributes() {
 
                 }
                 sortableAttributesMap.put(views[i].view_name, sortableAttributesList);
+                sortableAttributesMap.put(views[i].view_name, sortableAttributesList);
```
Comment (jaymcnallie) [inline catalog/nsili/catalog-nsili-source/src/main/java/com/connexta/alliance/nsili/source/NsiliSource.java]: Duplicate line of code?

File: catalog/nsili/catalog-nsili-transformer/src/main/java/com/connexta/alliance/nsili/transformer/DAGConverter.java [hunk 5]
```diff
@@ -1173,7 +1173,7 @@ public static void logMetacard(Metacard metacard, String id) {
                     String valueStr = getValueString(attribute.getValues());
                     LOGGER.trace("{} :  {} : {}", id, descriptor.getName(), valueStr);
                 } else {
-                    LOGGER.trace("{} : {} : {}", id, descriptor.getName(), attribute.getValue());
+                    LOGGER.trace("{} :  {} : {}", id, descriptor.getName(), attribute.getValue());
```
Comment (jaymcnallie) [inline catalog/nsili/catalog-nsili-transformer/src/main/java/com/connexta/alliance/nsili/transformer/DAGConverter.java]: Nit - extra space inserted here.  Surprised checkstyle didn't complain about this.

File: catalog/nsili/catalog-nsili-source/src/main/java/com/connexta/alliance/nsili/source/NsiliSource.java (hunk 4 above)
Comment (troymohl) [inline catalog/nsili/catalog-nsili-source/src/main/java/com/connexta/alliance/nsili/source/NsiliSource.java]: It was, and removed it.

File: catalog/nsili/catalog-nsili-transformer/src/main/java/com/connexta/alliance/nsili/transformer/DAGConverter.java (hunk 5 above)
Comment (troymohl) [inline catalog/nsili/catalog-nsili-transformer/src/main/java/com/connexta/alliance/nsili/transformer/DAGConverter.java]: Removed the extra space.

**Comment by jaymcnallie** [comment]
Built on a clean repo and unit tests passed.

**Pull request merged**